ducktape/tests/reporter.py

```diff
@@ -12,98 +12,131 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from ducktape.utils.terminal_size import get_terminal_size
+
 import json
 import os
 import shutil
 import pkg_resources
 
 
-class TestReporter(object):
-    def __init__(self, results):
-        """
-        :type results: ducktape.tests.result.TestResults
-        """
-        self.results = results
+DEFAULT_SEPARATOR_WIDTH = 100
 
-    def pass_fail(self, success):
-        """Convenient helper. Converts boolean to PASS/FAIL."""
-        return "PASS" if success else "FAIL"
 
-    def format_time(self, t):
-        """Return human-readable interval of time.
-        Assumes t is in units of seconds.
-        """
-        minutes = int(t / 60)
-        seconds = t % 60
+def pass_fail(success):
+    """Convenient helper. Converts boolean to PASS/FAIL."""
+    return "PASS" if success else "FAIL"
+
+
+def format_time(t):
+    """Return human-readable interval of time.
+    Assumes t is in units of seconds.
+    """
+    minutes = int(t / 60)
+    seconds = t % 60
+
+    r = ""
+    if minutes > 0:
+        r += "%d minute%s " % (minutes, "" if minutes == 1 else "s")
+    r += "%.3f seconds" % seconds
+    return r
+
+
+class SingleResultReporter(object):
+    """Helper class for creating a view of results from a single test."""
+
+    def __init__(self, result):
+        self.result = result
+        self.width = get_terminal_size()[0]
+
+    def result_string(self):
+        """Stringify single result"""
+        result_lines = [
+            "test_id:    %s" % self.result.test_context.test_id,
+            "status:     %s" % pass_fail(self.result.success),
+            "run time:   %s" % format_time(self.result.run_time),
+        ]
+
+        if self.result.data is not None:
+            result_lines.append(json.dumps(self.result.data))
```

Comment:
At this stage of the game without any automatic checks on the content of this data, I think it's better for the data collected to be immediately visible without extra effort.

Otherwise it is easy for "false positives" to slide by undiscovered as with tests using ConsumerPerformanceService